[Bugfix#281] - Refactor species handling incase of unknown species and enhance landing pages

ke4 opened the pull request:
Introduced a utility method for species name processing to ensure consistency across controllers. Updated species icon handling to use a default icon when an unknown species is encountered. Improved templates for baseline and differential experiments to handle empty datasets gracefully and prevent display issues.

ke4 committed:
Refactor species handling incase of unknown species and enhance landing pages

Introduced a utility method for species name processing to ensure consistency across controllers. Updated species icon handling to use a default icon when an unknown species is encountered. Improved templates for baseline and differential experiments to handle empty datasets gracefully and prevent display issues.

ke4 committed:
Refactor species name formatting into reusable methods.

Extracted species name formatting logic into dedicated utility methods in both controllers. This reduces code duplication and improves readability and maintainability.

amnonkhen commented on app/src/main/java/uk/ac/ebi/atlas/controllers/page/BaselineExperimentsController.java:
```diff
@@ -57,4 +60,8 @@ public String getBaselineExperimentsPage(Model model) {
         return "baseline-landing-page";
     }
+
+    private static String getSpeciesName(Experiment<?> experiment) {
```
I would put this method in the Experiment.

ke4 replied:
Make sense. I am going to do it.

amnonkhen commented on app/src/main/java/uk/ac/ebi/atlas/controllers/page/PlantExperimentsController.java:
```diff
@@ -65,4 +67,8 @@ else if (experiment.getType().isDifferential()) {
         return "plants-landing-page";
     }
+
+    private static String getSpeciesName(String speciesName) {
```
Similar to baseline controller's function.

ke4 replied:
Make sense. I am going to do it.

ke4 committed:
Refactor species name handling in controllers

Replaced inline species name normalization logic with existing dedicated methods (`getNormalisedSpeciesName` and `getNormalisedSpeciesReferenceName`). Remove the redundant ` getSpeciesName ` method from both controllers to enhance code maintainability and reduce duplication.

ke4 commented:
I am going to merge it as @amnonkhen already approved it and I also implemented his suggestions that he also approved.
